### ISSUE TITLE
Test playbooks on Ubuntu 21.04

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#2143]
Ubuntu 21.04 is due out this week. There is a beta available already. It would be good for us to verify whether the playbooks work properly on this new version of the OS, and also include it in the list of OSs deployed by the [DockerStatic role](https://github.com/AdoptOpenJDK/openjdk-infrastructure/tree/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic) (possibly replacing 20.10)

- [x] Check if the existing playbooks run on 21.04
- [x] Make any changes required for running on 21.04
- [x] Add to VagrantPlaybookCheck
- [x] Add to DockerStatic role

Details:

**Comment by Haroon-Khel** [comment]
First hurdle

<img width="794" alt="Screenshot 2021-04-28 at 18 43 03" src="https://user-images.githubusercontent.com/34969545/116449144-c40dfb00-a851-11eb-8672-f2eae9fab3e9.png">

**Comment by Haroon-Khel** [comment]
It also seems to have a problem with the NTP time role

<img width="712" alt="Screenshot 2021-04-29 at 12 13 35" src="https://user-images.githubusercontent.com/34969545/116542336-77273480-a8e4-11eb-9949-35b0aa7eaeed.png">

iirc, ubuntu 20.04 had this same problem

**Comment by Haroon-Khel** [comment]
ref https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1226

**Comment by sxa** [comment]
GCC role has a guard to stop it executing on Ubuntu 20 - needs to be modified to say >=20
https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/a4d8543253f2cf3edbe9b606b809a3ad5fd6beab/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Ubuntu.yml#L89

NTP may be similar but to clarify did we get a solution that didn't just involve disabling NTP on 20.04?

**Comment by Willsparker** [comment]
Can't you use `timesyncd` for Ubuntu?

**Comment by Haroon-Khel** [comment]
>NTP may be similar but to clarify did we get a solution that didn't just involve disabling NTP on 20.04?

https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1340

https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1226#issuecomment-634068286 is the reason why the `Set timedatectl set-ntp no` task of the NTP role fails

**Comment by sxa** [comment]
Yeah looks like another case where we've done `== 20` instead of `>=20`

**Comment by Haroon-Khel** [comment]
>Yeah looks like another case where we've done == 20 instead of >=20

Same with https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/a4d8543253f2cf3edbe9b606b809a3ad5fd6beab/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Ubuntu.yml#L84, as there isnt `gcc-4.8` or `g++-4.8` package available for ubuntu 2104

<img width="694" alt="Screenshot 2021-04-29 at 14 20 36" src="https://user-images.githubusercontent.com/34969545/116557161-156fc600-a8f6-11eb-8fab-c253feea17a2.png">

**Comment by Haroon-Khel** [comment]
Besides the gcc and ntp tasks, the rest of the playbook runs fine. Im running builds and tests on the machine to test if everything looks good

**Comment by Haroon-Khel** [comment]
@sxa Is there a reason the [OpenSSL102 role](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL102/tasks/main.yml) doesnt run on ubuntu? The [OpenSSL](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml) role does but it isnt included in the top level main.yml, and doesnt install the latest openssl

**Comment by sxa** [comment]
> @sxa Is there a reason the [OpenSSL102 role](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL102/tasks/main.yml) doesnt run on ubuntu? The [OpenSSL](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml) role does but it isnt included in the top level main.yml, and doesnt install the latest openssl

The OpenJ9 minimum version is 1.0.2. That role is only needed on distributions which have an earlier version included by default.

**Comment by Haroon-Khel** [comment]
Both openj9 and hotspot jdk 11 builds passed. A openjdk sanity test of jdk 11 hotspot ran sucessfully

**Comment by sxa** [comment]
Excellent news! We can leave @Willsparker to add it to VPC if he gets bored in the next couple of week ;-)

**Comment by Willsparker** [comment]
You're _so kind_ :joy:

**Comment by Willsparker** [comment]
Interesting issue with the Vagrantfile: 
```
11:14:36 + ssh -p 2202 -i /home/vagrant2/workspace/VagrantPlaybookCheck/OS/Ubuntu2104/label/vagrant/adoptopenjdkPBTests/willsparker-2143/ansible/id_rsa vagrant@127.0.0.1 'cd /vagrant/pbTestScripts && ./buildJDK.sh --branch master --fork AdoptOpenJDK --version jdk8 '
11:14:36 + tee /home/vagrant2/workspace/VagrantPlaybookCheck/OS/Ubuntu2104/label/vagrant/adoptopenjdkPBTests/logFiles/willsparker.2143.Ubuntu2104.build_log
11:14:38 bash: ./buildJDK.sh: /bin/bash: bad interpreter: Invalid argument
```
Not sure what the issue is, but it works when I try it on the VM manually, and it works when I run `ssh ...etc 'bash /vagrant/pbTestScripts/buildJDK.sh ...etc'` - I've got a commit in that does that instead, and tests all the other platforms to ensure it doesn't break anything - I doubt it will, but better safe than sorry: https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/1158/

**Comment by Willsparker** [comment]
With #2169 merged, the only thing left is adding it to the DockerStatic role, going off the list in the issue description.

**Comment by Haroon-Khel** [comment]
Ill work on adding ubuntu 21.04 to the DockerStatic role right now

**Comment by Willsparker** [comment]
I think this can be closed now, with #2189 in, right? :-)

**Comment by karianna** [comment]
> I think this can be closed now, with #2189 in, right? :-)

@sxa ^^

**Comment by Haroon-Khel** [comment]
Can be closed as https://github.com/adoptium/infrastructure/pull/2189 has been merged
